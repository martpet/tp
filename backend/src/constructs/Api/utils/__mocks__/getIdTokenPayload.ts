export const getIdTokenPayload = vi.fn().mockReturnValue({
  sub: 'dummySub',
  nonce: 'dummyNonce',
  aud: 'dummyAud',
  givenName: 'dummyGivenName',
  familyName: 'dummyFamilyName',
  picture: 'dummyPicture',
});
