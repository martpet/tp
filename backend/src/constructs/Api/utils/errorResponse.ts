import { APIGatewayProxyResult } from 'aws-lambda';
import { getReasonPhrase, StatusCodes } from 'http-status-codes';
import { SetRequired } from 'type-fest';

import { EnvName } from '~/types';

type ErrorResponseOptions = WithExposedError | WithoutExposedError;

type BaseOptions = {
  description?: string;
  error?: unknown;
  statusCode?: StatusCodes;
};

type WithExposedError = SetRequired<BaseOptions, 'error'> & {
  exposeErrorMessage: true;
};

type WithoutExposedError = BaseOptions & {
  exposeErrorMessage?: never | false;
};

const exposeErrorEnvs: EnvName[] = ['personal', 'staging'];

export const errorResponse = (
  traceId: string,
  {
    statusCode = StatusCodes.INTERNAL_SERVER_ERROR,
    description,
    error,
    exposeErrorMessage = exposeErrorEnvs.includes(global.cdkContextEnv),
  }: ErrorResponseOptions = {}
): APIGatewayProxyResult => {
  const bodyErrorObject: Record<string, unknown> = {
    statusCode,
    message: getReasonPhrase(statusCode),
    traceId,
  };

  if (description) {
    bodyErrorObject.description = description;
  }

  if (error instanceof Error) {
    console.error(`[${traceId}]`, error);

    if (exposeErrorMessage) {
      bodyErrorObject.error = String(error);
    }
  }

  return {
    statusCode,
    body: JSON.stringify({ error: bodyErrorObject }),
  };
};
