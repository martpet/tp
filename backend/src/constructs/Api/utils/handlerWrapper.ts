import { APIGatewayProxyHandlerV2 } from 'aws-lambda';

export const handlerWrapper =
  (func: APIGatewayProxyHandlerV2) =>
  (...params: Parameters<APIGatewayProxyHandlerV2>) =>
    func(...params);
