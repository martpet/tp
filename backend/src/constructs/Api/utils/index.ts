export * from './cookieName';
export * from './errorResponse';
export * from './getIdTokenPayload';
export * from './handlerWrapper';
export * from './itResolvesWithErrorResponse';
export * from './itResolvesWithLambdaEdgeErrorResponse';
export * from './lambdaEdgeErrorResponse';
export * from './parseEventCookies';
export * from './parseLambdaEdgeEventCookies';
