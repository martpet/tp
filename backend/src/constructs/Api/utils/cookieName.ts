import { ValueOf } from 'type-fest';

import { EndpointsCookies } from '~/constructs/Api/types';

export const cookieName = (name: ValueOf<EndpointsCookies>[number]) => name;
