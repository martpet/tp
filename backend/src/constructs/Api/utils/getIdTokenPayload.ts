import camelcaseKeys from 'camelcase-keys';

import { IdTokenPayload } from '~/constructs/Api/types';

export const getIdTokenPayload = (idToken: string): IdTokenPayload => {
  const payload = JSON.parse(Buffer.from(idToken.split('.')[1], 'base64').toString());

  return camelcaseKeys(payload);
};
