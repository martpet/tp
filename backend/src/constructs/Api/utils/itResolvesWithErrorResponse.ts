import { APIGatewayProxyHandlerV2 } from 'aws-lambda';

import { errorResponse } from '~/constructs/Api/utils';

export const itResolvesWithErrorResponse = (
  handler: APIGatewayProxyHandlerV2,
  args: Parameters<APIGatewayProxyHandlerV2>
) => {
  it('calls "errorResponse" with correct args', async () => {
    await handler(...args);
    expect(vi.mocked(errorResponse).mock.calls).toMatchSnapshot();
  });

  it('resolves with a correct value', () => {
    return expect(handler(...args)).resolves.toMatchSnapshot();
  });
};
