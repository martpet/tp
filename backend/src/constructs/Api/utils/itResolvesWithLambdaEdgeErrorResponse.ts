import { LambdaEdgeViewerRequestHandler } from '~/constructs/Api/types';
import { lambdaEdgeErrorResponse } from '~/constructs/Api/utils';

export const itResolvesWithLambdaEdgeErrorResponse = (
  handler: LambdaEdgeViewerRequestHandler,
  args: Parameters<LambdaEdgeViewerRequestHandler>
) => {
  it('calls "lambdaEdgeErrorResponse" with correct args', async () => {
    await handler(...args);
    expect(vi.mocked(lambdaEdgeErrorResponse).mock.calls).toMatchSnapshot();
  });

  it('resolves with a correct value', () => {
    return expect(handler(...args)).resolves.toMatchSnapshot();
  });
};
