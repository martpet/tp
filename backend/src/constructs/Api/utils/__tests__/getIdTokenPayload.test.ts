import { getIdTokenPayload } from '~/constructs/Api/utils/getIdTokenPayload';

const args = ['.eyJkdW1teV9pZHRva2VuX2tleSI6ImR5bW15X2lkdG9rZW5fdmFsIn0=.'] as const;

describe('getIdTokenPayload', () => {
  it('returns a correct value', () => {
    expect(getIdTokenPayload(...args)).toMatchSnapshot();
  });
});
