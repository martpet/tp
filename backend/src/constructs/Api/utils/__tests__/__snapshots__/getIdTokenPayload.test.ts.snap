// Vitest Snapshot v1

exports[`getIdTokenPayload > returns a correct value 1`] = `
{
  "dummyIdtokenKey": "dymmy_idtoken_val",
}
`;
